contrib/core-contract-tests/contracts/pox/pox-4.clar

```diff
@@ -0,0 +1,1355 @@
+;; PoX testnet constants
```

Comment:
This belongs in `stackslib/src/chainstate/stacks/boot/pox-4.clar`

Reply:
It is part of `stackslib/src/chainstate/stacks/boot/pox-testnet.clar`

Reply:
What I mean is, this file should not exist.  It's a duplicate of code in `stackslib/src/chainstate/stacks/boot/pox-4.clar`.  There should only be one PoX-4 code body in the repository.

Reply:
I have added a test that ensure that the duplicated code is always the same. Hopefully, we can remove it once clarity can handle this better.

Reply:
I'm sorry, but I must insist that we only have one copy of the PoX-4 code body in the repo.

Reply:
@friedger I concur with Jude here - keeping two copies of a boot contract adds an extra burden to keep them both up to date at a minimum. 
could you use the contract(s) in `./stackslib/src/chainstate/stacks/boot`, or possibly create a symlink in the `./contrib` dir?

Reply:
> I have added a test that ensure that the duplicated code is always the same. Hopefully, we can remove it once clarity can handle this better.

another option rather than a symilnk - wouldn't copying the existing contract to an ephemeral location also work (provided the test removes that ephemeral contract upon conclusion)?

Reply:
@jcnelson The two files are different, therefore the two different names.

Reply:
I created an issue on clarinet: https://github.com/hirosystems/clarinet/issues/1267 Do we want to have this blocked PR blocked until it is released? Can we afford that?

Alternatively, we can remove clarinet tests and commit a change to a boot contract without tests. Does not look good to me. 

I take ownership of the situation with clarinet testing of pox-4 and try to improve the situation as soon as possible. To me, it is not a blocker.

Reply:
`pox-testnet.clar + pox-4.clar = pox-4-testnet.clar`
Clarinet unit tests can't run on pox-4.clar because that file does not define a valid contract. Instead we create pox-4-testnet.clar and run the unit tests against the new, valid contract file. (pox-testnet/pox-mainnet.clar only consists of constant definitions.)

Hugo suggested on [a different issue](https://github.com/stacks-network/stacks-core/pull/4106#issuecomment-1836655726) to make pox-4.clar a valid contract and use `is-in-mainnet` in that file. If @jcnelson  agrees, I am happy to update this PR and remove the current situation.